modules/core/arrow-core-data/src/main/kotlin/arrow/typeclasses/Traverse.kt

```diff
@@ -5,11 +5,519 @@ import arrow.core.Id
 import arrow.core.identity
 import arrow.core.value
 import arrow.typeclasses.internal.IdBimonad
+import arrow.core.Option
+import arrow.core.Either
+import arrow.core.Validated
+import arrow.core.Const
+import arrow.core.Nel
+import arrow.core.SequenceK
 
 /**
  * ank_macro_hierarchy(arrow.typeclasses.Traverse)
  *
- * Traverse, also known as Traversable. Traversal over a structure with an effect.
+ * In functional programming it is very common to encode "behaviors" as data types - common behaviors include [Option] for possibly missing values, [Either] and [Validated] for possible errors, and [Ref]({{ '/docs/effects/ref/' | relative_url }}) for asynchronous and concurrent access and modification of its content.
+ *
+ * These behaviors tend to show up in functions working on a single piece of data - for instance parsing a single [String] into an [Int], validating a login, or asynchronously fetching website information for a user.
+ *
+ * ```kotlin:ank
+ * import arrow.fx.IO
+ *
+ * //sampleStart
+ * interface Profile
+ * interface User
+ *
+ * fun userInfo(u: User): IO<Profile>
+ * //sampleEnd
+ * ```
+ *
+ * Each function asks only for the data it needs; in the case of `userInfo`, a single `User`. Indeed, we could write one that takes a `List<User>` and fetches profile for all of them, but it would be a bit strange. If we just wanted to fetch the profile of only one user, we would either have to wrap it in a [List] or write a separate function that takes in a single user, nonetheless. More fundamentally, functional programming is about building lots of small, independent pieces and composing them to make larger and larger pieces - does it hold in this case?
+ *
+ * Given just `(User) -> IO<Profile>`, what should we do if we want to fetch profiles for a `List<User>`? We could try familiar combinators like map.
+ *
+ * ```kotlin:ank
+ * fun profilesFor(users: List<User>): List<IO<Profile>> = users.map(::userInfo)
+ * ```
+ *
+ * Note the return type `List<IO<Profile>>`. This makes sense given the type signatures, but seems unwieldy. We now have a list of asynchronous values, and to work with those values we must then use the combinators on `IO` for every single one. It would be nicer instead if we could get the aggregate result in a single `IO`, say a `IO<List<Profile>>`.
+ *
+ * ### Sequencing
```

Comment:
@pakoito and @JorgeCastilloPrz suggested to start first with Sequence and then explain `traverse`. in that sense, I reverted the former structure. Maybe we can add something tangible in the Text, to add guide the reader towards a redline. If you have something in mind, feel free to comment it :)

Reply:
It is a good idea to start with `sequence`, but not by explaining it as `traverse` + `id`. I'd say don't mention `traverse` at all up until you introduce `traverse` and then quickly mention how `sequence` is `traverse(AP, ::identity)`

Reply:
I think that is one bargain we have to take when introducing sequence first. It's inconclusive if we introduce sequence without explaining it and still delaying `traverse`. For what its worth I suggest a different order.  Where we start with Sequence then Traverse and then Sequence Effects. But I am not sure if that changes anything.

Reply:
Starting with sequence is a really good starting point because most who are stuck and need traverse are sitting on a type `Kind<F, Kind<G, A>>` already and just need `sequence` to solve their problem. Then transition into the more general form `traverse`, show that `traverse(::identity) == sequence` and `map(f).sequence() == traverse(f)`. After that the rest of the tutorial around sequencing for effects, foldable, etc. That is how I would structure it.